File: 16_spring_cache/local_cache/src/main/java/com/yologger/local_cache/controller/MemberController.java
package com.yologger.local_cache.controller;

import com.yologger.local_cache.service.MemberDTO;
import com.yologger.local_cache.service.MemberService;
import lombok.RequiredArgsConstructor;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

import java.util.List;

@RestController
@RequestMapping("/member")
@RequiredArgsConstructor
public class MemberController {
    private final MemberService memberService;

    @GetMapping
    public ResponseEntity<List<MemberDTO>> getAll() {
        return new ResponseEntity<>(memberService.getAllUsers(), HttpStatus.OK);
    }

    @GetMapping("/{id}")
    public ResponseEntity<MemberDTO> findMemberById(@PathVariable Long id) {
        return new ResponseEntity<>(memberService.findMemberById(id), HttpStatus.OK);
    }

    @PutMapping
    public ResponseEntity<Void> updateEmailById(@RequestBody UpdateEmailByIdRequest request) {
        memberService.updatePassword(request.getId(), request.getPassword());
        return ResponseEntity.noContent().build();
    }

    @DeleteMapping("/{id}")
    public ResponseEntity<Void> deleteMemberById(@PathVariable Long id) {
        memberService.deleteMemberById(id);
        return ResponseEntity.noContent().build();
    }
}

File: 21_spring_webflux/02_webflux_kotlin/webflux_kotlin/src/main/kotlin/com/yologger/samples/webflux_kotlin/controller/MemberController.kt
package com.yologger.samples.webflux_kotlin.controller

import com.yologger.samples.webflux_kotlin.service.MemberDTO
import com.yologger.samples.webflux_kotlin.service.MemberService
import kotlinx.coroutines.flow.Flow
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.http.HttpStatus
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.PathVariable
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/member")
class MemberController constructor(
    @Autowired val memberService: MemberService
){
    @PostMapping
    suspend fun addMember(@RequestBody request: AddMemberRequest): ResponseEntity<Long> {
        return ResponseEntity(memberService.addMember(MemberDTO(email = request.email, name = request.name, password = request.password)).id!!, HttpStatus.CREATED)
    }

    @GetMapping("/{id}")
    suspend fun findMemberById(@PathVariable("id") id: Long): ResponseEntity<MemberDTO> {
        return ResponseEntity(memberService.findMemberById(id), HttpStatus.OK)
    }

    @GetMapping("")
    suspend fun findAllMembers(): ResponseEntity<Flow<MemberDTO>> {
        return ResponseEntity(memberService.findAllMember(), HttpStatus.OK)
    }
}